src/macro_py/MacroRecorder.py

```
@@ -523,6 +531,19 @@ def on_scroll(self, x, y, dx, dy):
     def on_key_press(self, key):
         try:
             if self.recording:
+                # Intercept stop hotkey (F2) as a control event (non-macOS in-process)
+                try:
+                    if key == keyboard.Key.f2:
+                        # Signal the GUI via a synthetic control event
+                        try:
+                            timestamp = time.time() - (self.start_time or time.time())
+                        except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)